### PR TITLE
Enabling trouter notifications for 1P app

### PR DESCRIPTION
**Pull request by prram**
This PR will enable all the 1p app to listen to trouter notifications using the trouter connection from teams client.

### CODE REVIEW

File: src/public/interfaces.ts [hunk 1]
```diff
@@ -483,3 +483,15 @@ export interface FrameContext {
    */
   websiteUrl: string;
 }
+
+export interface NotificationPayload {
```
Comment (raulfernandes) [inline src/public/interfaces.ts:487]: Rename to TrouterNotificationPayload

File: test/private/notifications.spec.ts [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+  });
+
+  afterEach(() => {
+    // Reset the object since it's a singleton
+    if (_uninitialize) {
+      _uninitialize();
+    }
+  });
+
+  describe('registerTrouterNotificagions', () => {
```
Comment (raulfernandes) [inline test/private/notifications.spec.ts:28]: Typo registerTrouterNotifications

File: test/private/notifications.spec.ts [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+import { Utils } from '../utils';
+import { _uninitialize } from '../../src/public/publicAPIs';
+import { NotificationPayload } from '../../src/public/interfaces';
+
+describe('logs', () => {
+  // Use to send a mock message from the app.
+  const utils = new Utils();
+  const notificationPayload: NotificationPayload =  {
+    payload : "{ \"eventId\" : 1500,  \"environment\" : \"Production\"}", 
+    subPath : 'shifts'
```
Comment (raulfernandes) [inline test/private/notifications.spec.ts:11]: Is this '/shifts' or just 'shifts'?

Comment (prram) [reply test/private/notifications.spec.ts:11]: its actually /shifts..I'll correct it although its in tests.

File: src/private/notifications.ts [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+ * The notification payload received is passed down to the app.
+ *
+ * @private
+ * Hide from docs
+ */
+export namespace notifications {
+  GlobalVars.handlers['trouter.notificationRecieved'] = handleProcessNotifications;
+
+  function handleProcessNotifications(payload: TrouterNotificationPayload): void {
+    if (GlobalVars.notificationHandler) {
```
Comment (sachingupta) [inline src/private/notifications.ts:16]: we are registering these kind of handlers inside internaleAPIs,ts 
image

Lets follow consistent approach

**Comment by TrevorJoelHarris** [comment]
This PR is stale and will be closed, please feel free to re-open it if your intention is still to check into main.

**Pull request closed without merging**